reginald/slack_bot/bot.py

```diff
@@ -382,50 +382,95 @@ async def _process_request(
         # start processing the message
         logging.info(f"Processing message '{message}' from user '{user_id}'.")
 
+        # remove clock as we're about to process the message
         await client.web_client.reactions_remove(
             name="clock2",
             channel=event["channel"],
             timestamp=event["ts"],
         )
 
-        # if this is a direct message to Reginald...
-        if event_type == "message" and event_subtype is None:
-            await self.react(client, event["channel"], event["ts"])
-            model_response = await asyncio.get_running_loop().run_in_executor(
-                None,
-                lambda: requests.get(
-                    f"{self.api_url}/direct_message",
-                    json={"message": message, "user_id": user_id},
-                ),
-            )
-
-        # if @Reginald is mentioned in a channel
-        elif event_type == "app_mention":
-            await self.react(client, event["channel"], event["ts"])
-            model_response = await asyncio.get_running_loop().run_in_executor(
-                None,
-                lambda: requests.get(
-                    f"{self.api_url}/channel_mention",
-                    json={"message": message, "user_id": user_id},
-                ),
-            )
+        try:
+            # try to get a response from the model api
+            # if this is a direct message to Reginald...
+            if event_type == "message" and event_subtype is None:
+                await self.react(client, event["channel"], event["ts"])
+                model_response = await asyncio.get_running_loop().run_in_executor(
+                    None,
+                    lambda: requests.get(
+                        f"{self.api_url}/direct_message",
+                        json={"message": message, "user_id": user_id},
+                    ),
+                )
 
-        # otherwise
-        else:
-            logging.info(f"Received unexpected event of type '{event['type']}'.")
-            return
+            # if @Reginald is mentioned in a channel
+            elif event_type == "app_mention":
+                await self.react(client, event["channel"], event["ts"])
+                model_response = await asyncio.get_running_loop().run_in_executor(
+                    None,
+                    lambda: requests.get(
+                        f"{self.api_url}/channel_mention",
+                        json={"message": message, "user_id": user_id},
+                    ),
+                )
 
-        if model_response.status_code != 200:
-            raise ValueError("Unable to get response.")
-        model_response = model_response.json()
+            # otherwise
+            else:
+                logging.info(f"Received unexpected event of type '{event['type']}'.")
+                return
+
+            if model_response.status_code != 200:
+                raise ValueError("Unable to get response.")
```

Comment:
Do we still need this bit?

I guess there are two scenarios here: 1) If the VM is off (gives connection timeout error) and 2) If the VM is on but for some reason the model api isn't working (what is returned error here?). 

Either way maybe this would be good to become a logged error too, something like:
```
if model_response.status_code != 200:
    err = f"Unable to get response. Returned response code: {model_response.status_code}"
    logging.error(err)
    raise ValueError(err)
```

Also (I know I was the one who put ValueError but) maybe it should be ``model_response.raise_for_status()`` as I think this just raises the error.

Reply:
good point! I think actually we do need something to raise an error so that we can catch it and log it. I think we could just replace all of this snippet with `model_response.raise_for_status`?

Reply:
so if there is an error, we jump to the `except Exception as err:` block and log it and create an appropriate response